### PR TITLE
Enable Incremental Adoption of Next.js 13.4+ App Router

### PR DESCRIPTION
**Pull request by doug-andrade**
This pull request introduces incremental adoption of the new Next.js 13.4+ app router to the Documenso project. The changes have been thoroughly tested and are compatible with the current app, ensuring a seamless transition.

### Summary of Changes:

- Configured the Next.js 13.4+ app router for incremental adoption.
- Tested the changes by running `npm build` and confirmed 1:1 compatibility with the existing app.
- Ensured proper integration of the new router while maintaining backward compatibility.

This update provides several benefits, including improved performance, enhanced routing capabilities, and better support for modern web development practices. By adopting the latest Next.js app router, we can take advantage of its features while minimizing disruptions to the existing codebase.

Please review these changes and consider merging them into the main branch. I'm confident that this update will contribute positively to the project's functionality and future development.

Thank you for your attention and consideration.

### CODE REVIEW

File: .env.example
Comment (doug-andrade) [inline .env.example:1]: This file needed to be moved to the `apps/web` directory. I also believe it make for sense to be in this directory.

File: apps/web/components/login.tsx
Comment (doug-andrade) [inline apps/web/components/login.tsx:1]: Tested `useRouter` and `useSearchParams`  from `next/navigation`  while still being in the pages directory. It works.

File: apps/web/pages/500.jsx
Comment (doug-andrade) [inline apps/web/pages/500.jsx:1]: Removed unused import.

File: apps/web/env.mjs
Comment (doug-andrade) [inline apps/web/env.mjs:1]: implemented type safe environment variables using `@t3-oss/env-core` , validations with zod

File: apps/web/next.config.js
Comment (doug-andrade) [inline apps/web/next.config.js:1]: simplified `next.config.js` by moving from the `next-transpile-modules` dependency to `transpilePackages` within the `nextConfig` object.

File: apps/web/package.json
Comment (doug-andrade) [inline apps/web/package.json:1]: rm >> `next-transpile-modules`
add >> `@t3-oss/env-core`,`zod`
update >> `next`

File: apps/web/.env.example
Comment (doug-andrade) [inline apps/web/.env.example:1]: .env.example was moved to apps/web to play nice with 13.4. Otherwise, the environment variables wouldn't be loaded.

**Comment by Mythie** [comment]
Hey Doug, just letting you know I'll be looking at this soon 😄

**Pull request closed without merging**